content/5_Plat/Sweep_Line.mdx

```diff
@@ -162,8 +162,165 @@ For each point, we recalculate $\delta = \min(\delta, \delta_p)$, and update our
 <FocusProblem problem="segSam" />
 
 ### Solution
+Let's simplify the problem a little bit and focus on finding any overlapping segments.
 
-<IncompleteSection />
+To find a pair of overlapping segments, use a sweep line approach by sweeping a vertical line across the scene from left to right, pausing at every segment endpoint.
+
+We simulate this by sorting all the segment endpoints by $x$ and walking through the sorted array (called 'events' in the code below). As we scan, we keep track of active segments
+using a set (called 'active_segments' in the code below). When we hit the beginning point of a segment, add it to the active set, and remove it from the active set when we hit the ending point of a segment.
+
+Inserting or removing the active segments from the set takes $\mathcal{O}(\log n)$ per operation.
+
+The active set of segments is ordered by $y$ coordinate. If two segments overlap, they are adjacent in the set, so every time we insert or remove a segment, we check if the adjacent segments overlap.
+
+Here is an animation of how it works:
+
+<video width="960" height="720" controls>
+	<source src="/animations/cowsteeplechase.mp4" type="video/mp4"/>
+</video>
+
+
+## Implementation
+
+Time complexity: $\mathcal{O}(n\log n)$.
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+long long sweep_line_x;
+
+// BeginCodeSnip{Point structure}
+struct Point {
+	long long x, y, segment_idx;
+	bool operator<(const Point &other) {
+		return x == other.x ? y < other.y : x < other.x;
+	}
+};
+// EndCodeSnip
+
+struct Segment {
+	long long a, b, x, y, idx;
+};
+
+double coordinate(Segment a) {
+	if (a.a == a.x) { return a.b; }
+	return a.b + (a.y - a.b) * (sweep_line_x - a.a) / (a.x - a.a);
+}
+
+int sign(long long x) {
+	if (x == 0) {
+		return 0;
+	} else {
+		return x < 0 ? -1 : +1;
+	}
+}
+
+int operator*(Point p1, Point p2) { return sign(p1.x * p2.y - p1.y * p2.x); }
+Point operator-(Point p1, Point p2) {
+	Point p = {p1.x - p2.x, p1.y - p2.y};
+	return p;
+}
+bool operator<(Segment a, Segment b) {
+	return a.idx != b.idx && coordinate(a) <= coordinate(b);
+}
+bool operator==(Segment a, Segment b) { return a.idx == b.idx; }
+
+// To check if two segments intersect we will use the signed area of the ABC
+// triangle. This can be derived from the cross product of the vectors AB and
+// AC.
```

Comment:
multiline comment

Reply:
wasn't fixed 👀

Reply:
was it?

Reply:
You know what, I'm gonna delete these lines.

Reply:
ayo what

Reply:
no like just even it out like this
```cpp
/*
 * To check if two segments intersect we will use the
 * signed area of the ABC triangle. This can be derived
 * from the cross product of the vectors AB and AC.
 */
```